docs/design/latency_definitions/index.md

```diff
@@ -0,0 +1,141 @@
+In this way, information is propagated from the sensor node to the actuator node.
+
+CARET measures the following three main locations for latency calculation.
+
+- Callback start
+- Callback end
+- Publish
+
+Latency is calculated by calculating the time difference required for each measurement section.
+
+For a more detailed definition, see
+
+- [Callback](./callback.md)
+- [Communication](./communication.md)
+- [Node](./node.md)
+- [Path](./path.md)
+
+CARET can also check other time series data on the Publish side as well.
+All the objects for which time series data can be obtained are listed below.
+
+| Target                              | Configuration required? |
+| ----------------------------------- | ----------------------- |
+| [Path](./path.md)                   | Yes                     |
+| [NodePath](./node.md)               | Yes                     |
+| [Communication](./communication.md) | No                      |
+| [Callback](./callback.md)           | No                      |
+| [Publisher](./publisher.md)         | No                      |
+| [Subscription](./subscription.md)   | No                      |
+| [Timer](./timer.md)                 | No                      |
+
+Here, for Path and NodePath, definitions must be given manually.
+For details on setting the definitions, see [Configuration](../../configuration/index.md).
+
+## Detailed Sequence
+
+Below is a detailed sequence diagram of the SingleThreadedExecutor, from publish in the callback to the execution of the subscription callback.
+
+```plantuml
+@startuml
+title: Definition of major tracepoints
+
+participant UserCode
+participant ROS2
+participant DDS
+participant LTTng
+
+== Publisher Side ==
+
+activate ROS2
+activate UserCode
+UserCode -> ROS2: publish()
+activate ROS2
+ROS2 -> LTTng: [rclcpp_publish] \n[rclcpp_intra_publish]
+
+
+ROS2 -> DDS: dds_write()
+deactivate ROS2
+activate DDS
+DDS -> LTTng: [dds_write]
+
+UserCode -> ROS2 : callback_end
+deactivate UserCode
+deactivate ROS2
+deactivate DDS
+
+
+== Subscription Side ==
+
+UserCode -> ROS2 : spin()
+activate DDS
+activate ROS2
+
+loop
+    ROS2 -> ROS2 : wait until next event
+    activate ROS2
+    DDS -> ROS2: <notify> on_data_available
+    ROS2 -> LTTng : [on_data_available]
+    deactivate ROS2
+
+    deactivate DDS
+    ROS2 -> DDS : reset ready-set, take messages
+    activate DDS
+    DDS -> ROS2
+    deactivate DDS
+
+    group execute timer callbacks
+    end
+
+    group execute subscription callbacks
+        ROS2 -> LTTng: [callback_start]
+        ROS2 -> UserCode: callback start
+        activate UserCode
+        UserCode -> ROS2: callback end
+        deactivate UserCode
+        ROS2 -> LTTng: [callback_end]
+    end
+
+    group execute service callbacks
+    end
+
+    group execute client callbacks
+    end
+end
+
+deactivate ROS2
+@enduml
+```
+
+Here, each element indicates the following
+
+- UseCode is a callback
```

Comment:
```suggestion
- UserCode is a callback
```

Reply:
fixed in :[b104708](https://github.com/tier4/CARET_doc/pull/106/commits/b104708b0378c76f80d2ba457892ff9ad0dbfdc4)